### ISSUE TITLE
Weird messages  when using the Shockwave generator CBM

### ISSUE THREAD

**Issue by John-Candlebury** [CleverRaven/Cataclysm-DDA#13693]
It tells you the duration the target will be stunned for. On a related note it it also literally tells you that the "%s was stunned AGAIN for %d turns!" if the target was stunned more than once and also provides.

Looks like they were either debug  messages that were never removed or that we stopped telling stun durations to the players long ago and didn't update this instance.

**Comment by ejseto** [comment]
Or I liked the idea of verbose descriptions when I made it. What's wrong with that?

**Comment by John-Candlebury** [comment]
Nothing, it's just that the game doesn't reveal that sort of information about stuns anywhere else. So it seems pretty weird.

**Comment by Rivet-the-Zombie** [comment]
I'm leaning towards less verbosity when it comes to this sort of situation; same as with enemies' current health and such, there's some information that the player shouldn't have handed directly to them.

**Comment by vache** [comment]
At the very least, it should convert from turns to a unit of time.  Though even better would be to split it into ranges and show a message depending on the range.

**Comment by swwu** [comment]
I think it's not so much as saying the duration, as it is knockback being the only one of the possible stun applications where the duration is mentioned (and the only one where "AGAIN" is mentioned). I'm happy to un-specific-ize this if it seems reasonable to people.

**Comment by Rivet-the-Zombie** [comment]
>I'm happy to un-specific-ize this if it seems reasonable to people.

It's got my vote.

**Comment by Coolthulhu** [comment]
I don't think monster stun needs to be announced in any way.

By the way, the whole `game::knockback` function (used by shockwave) is horribly ugly and should be reduced to 30% of its current size. Or even replaced completely by the fling creature function.

**Comment by swwu** [comment]
PR'd: #13778

**Comment by swwu** [comment]
@Coolthulhu I agree 100% with the `fling_creature` suggestion. It looks like the `if (c.monster?) {c.foo()} else if (c.npc?) {c.bar()}` pattern is everywhere again and that would be a change against it.

**Comment by Night-Pryanik** [comment]
Should be closed after #13778?
